python/ray/node.py

```diff
@@ -639,10 +652,47 @@ def start_raylet(self, use_valgrind=False, use_profiler=False):
         assert ray_constants.PROCESS_TYPE_RAYLET not in self.all_processes
         self.all_processes[ray_constants.PROCESS_TYPE_RAYLET] = [process_info]
 
+    def get_job_redirected_log_file(self, worker_id, job_id):
```

Comment:
Agreed that there's a lot of files, but we generate a lot of logs. It's complicated because worker, job, redis, raylet, etc logs all share parts of this code path.

Reply:
Sorry, I didn't mean a lot of files, I meant a lot of **functions** (in the code) to open files, and it's unclear which one is responsible for what (or why opening a log requires so many functions). Ideally there would be 1 function responsible for opening any log file, though that's probably unrealistic. But can we simplify it from the 6+ we have now?